shared-libs/cht-datasource/src/libs/contact.ts

```diff
@@ -0,0 +1,16 @@
+import { Doc } from './doc';
+import { DataObject } from './core';
+
+interface DehydratedParent extends DataObject {
+  _id: string;
+  parent?: DehydratedParent;
+}
+
+/** @internal */
+export interface Contact extends Doc {
+  contact_type?: string;
+  name?: string;
+  parent?: DehydratedParent;
+  reported_date?: Date;
+  type: string;
```

Comment:
note: A heads-up on this. `type` is a keyword in Typescript but this is allowed and in the future, if we decide to raise strictness this will raise an error at some point.

Reply:
Good point!  We need this as `type` on v1 for backwards compatibility, but on future iterations, we should avoid keywords like this.